app/main.py

```diff
@@ -2,0 +1,29 @@
+class Animal:
+    alive = []
+
+    def __init__(self, name: str, health: int = 100) -> None:
+        self.name = name
+        self.health = health
+        self.hidden = False
+        Animal.alive.append(self)
+
+    def __repr__(self) -> str:
+        return (f"{{Name: {self.name}, "
+                f"Health: {self.health}, Hidden: {self.hidden}}}")
+
+    def check_alive(self) -> None:
+        if self.health <= 0 and self in Animal.alive:
+            Animal.alive.remove(self)
```

Comment:
The `check_alive` method correctly removes the animal from the `alive` list if its health is zero or less. Ensure that this logic is consistent with the task requirements for managing animal states.